packages/server/modules/core/migrations/20250123162109_required_indices.ts

```diff
@@ -0,0 +1,92 @@
+import { Knex } from 'knex'
+
+export async function up(knex: Knex): Promise<void> {
+  await knex.schema.alterTable('stream_commits', (table) => {
+    // table.index('streamId') // index should already be available due to this being the first item in the composite primary key
+    table.index('commitId')
+  })
+  await knex.schema.alterTable('branch_commits', (table) => {
+    // table.index('branchId') // index should already be available due to this being the first item in the composite primary key
+    table.index('commitId')
+  })
+  await knex.schema.alterTable('refresh_tokens', (table) => {
+    // table.index('id') // index should already be available due to this being the primary key
+    table.index('userId')
+  })
+  await knex.schema.alterTable('commits', (table) => {
+    // table.index('id') // index should already be available due to this being the primary key
+    table.index('author')
```

Comment:
do these really need to be separate indices, instead of composite ones? it was at least the case in MySQL that if you had an index like "x, y, z", then it would be used even if you filter by only "x" or "x" and "y", as long as the order is the same (so it wouldn't work when just filtering by "z")

Reply:
important to think about this, cause they do take up space

Reply:
That's even better than what I expected then, I didn't think there would be an index on just "y" at all. But my point is that we need to look at what the actual filter conditions are, and if "y" is usually used together with "x", then there's no need to make "y" have its own index

Reply:
Just having X and Y as separate indices could even be worse than having a "X, Y" index, if both are filtered together. It's just going to use the X or the Y index, where "X, Y" would be the better one